hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/index/IndexResolver.java

```diff
@@ -136,20 +174,101 @@ private IndexResolver() {
         }
 
         List<RelNode> rels = new ArrayList<>(supportedIndexes.size());
-
         for (MapTableIndex index : supportedIndexes) {
             // Create index scan based on candidates, if possible. Candidates could be merged into more complex
             // filters whenever possible.
-            RelNode rel = createIndexScan(scan, distribution, index, conjunctions, candidates);
+            List<Boolean> ascs = buildFieldDirections(index, true);
+            RelNode relAscending = createIndexScan(scan, distribution, index, conjunctions, candidates, ascs);
+
+            if (relAscending != null) {
+                RelCollation relAscCollation = getCollation(relAscending);
+                if (fullScanRelsMap.containsKey(relAscCollation)) {
```

Comment:
A call to `containsKey` is not needed, since `remove` already does this check.

Reply:
Optimized in https://github.com/hazelcast/hazelcast/pull/18013/commits/341d45d5e66304f040edb42aab1202596c2b4646